### PR TITLE
Fix channel name size to not overlap other views

### PR DESCRIPTION
**Pull request by JcMinarro**
### Description
Fix channel name size to not overlap other views

| Before | After |
| --- | --- |
|  photo_2021-01-13_14-29-07  |  photo_2021-01-13_14-29-10 |
| photo_2021-01-13_14-29-08  |  photo_2021-01-13_14-29-09 |
### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- ~[ ] Changelog updated with client-facing changes~
- ~[ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

**Review by zsmb13 — COMMENTED** [review]
Don't we want to ellipsize these to let the user know that there's more content there that we couldn't display?

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_messages_header_view.xml [hunk 1]
```diff
@@ -48,13 +48,15 @@
 
     <TextView
         android:id="@+id/title"
-        android:layout_width="wrap_content"
+        android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:layout_marginTop="8dp"
         android:lineSpacingExtra="5sp"
+        android:gravity="center_horizontal"
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/res/layout/stream_ui_messages_header_view.xml:55]: Will `android:letterSpacing="0"` with ellipsize work for us?

**Review by bychkovdmitry — APPROVED** [review]

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_messages_header_view.xml (hunk 1 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_messages_header_view.xml:55]: I think we should change `  <style name="StreamUiTextAppearance" parent="TextAppearance.MaterialComponents.Body2" />` to `  <style name="StreamUiTextAppearance" parent="TextAppearance.AppCompat.Body2" />` which doesn't apply `letterSpacing` to fix ellipsize issue in other places

**Review by adasiewiczr — APPROVED** [review]

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_channel_list_item_foreground_view.xml [hunk 2]
```diff
@@ -21,10 +21,12 @@
...
     <TextView
         android:id="@+id/channel_name_label"
-        android:layout_width="wrap_content"
+        android:layout_width="0dp"
         android:layout_height="wrap_content"
-        android:layout_marginStart="8dp"
+        android:layout_marginHorizontal="8dp"
         android:textAppearance="@style/StreamUiTextAppearance.BodyBold"
+        android:maxLines="1"
+        app:layout_constraintEnd_toEndOf="parent"
```
[the remark this answers is not in the document]
Comment (zsmb13) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_channel_list_item_foreground_view.xml:30]: I'll fix this in a follow up, that badge has other work to be done on it too

**Pull request merged**